flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/kmeans/KMeansModelData.java

```diff
@@ -47,12 +52,69 @@
 
     public DenseVector[] centroids;
 
-    public KMeansModelData(DenseVector[] centroids) {
+    /**
+     * The weight of the centroids. It is used when updating the model data in online training
+     * process.
+     *
+     * <p>KMeansModelData objects generated during {@link KMeans#fit(Table...)} also contains this
+     * field, so that it can be used as the initial model data of the online training process.
+     */
+    public DenseVector weights;
+
+    public KMeansModelData(DenseVector[] centroids, DenseVector weights) {
+        Preconditions.checkArgument(centroids.length == weights.size());
         this.centroids = centroids;
+        this.weights = weights;
     }
 
     public KMeansModelData() {}
 
+    /**
+     * Generates a Table containing a {@link KMeansModelData} instance with randomly generated
+     * centroids.
+     *
+     * @param env The environment where to create the table.
+     * @param k The number of generated centroids.
+     * @param dim The size of generated centroids.
+     * @param weight The weight of the centroids.
+     * @param seed Random seed.
+     */
+    public static Table generateRandomModelData(
+            StreamExecutionEnvironment env, int k, int dim, double weight, long seed) {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        return tEnv.fromDataStream(
+                env.fromElements(1).map(new RandomCentroidsCreator(k, dim, weight, seed)));
+    }
+
+    private static class RandomCentroidsCreator implements MapFunction<Integer, KMeansModelData> {
+        private final int k;
+        private final int dim;
+        private final long seed;
+        private final double weight;
+
+        private RandomCentroidsCreator(int k, int dim, double weight, long seed) {
+            this.k = k;
+            this.dim = dim;
+            this.seed = seed;
```

Comment:
nits: maybe reorder this line with the line below for consistency with the method's input parameters?

Same for the member private variable declarations :)